### PR TITLE
Respect the down domain's support for automatic local access

### PR DESCRIPTION
**Pull request by e-kayrakli**
https://github.com/chapel-lang/chapel/pull/17009 added support for automatic
local access using rank-change views. However, that support didn't care whether
base domain supported that optimization. This PR fixes that.

Test:
- [x] gasnet `-sdistType=DistType.replicated` in `distributions/robust/arithmetic`
- [x] gasnet in `optimizations`

### CODE REVIEW

**Comment by e-kayrakli** [comment]
@bradcray -- just FYI. A trivial fix for the PR that you have reviewed.

File: modules/internal/ArrayViewRankChange.chpl [hunk 1]
```diff
@@ -429,7 +429,7 @@ module ArrayViewRankChange {
     }
 
     override proc dsiSupportsAutoLocalAccess() param {
-      return true;
+      return upDom.dsiSupportsAutoLocalAccess();
```
Comment (e-kayrakli) [inline modules/internal/ArrayViewRankChange.chpl:432]: whoops, should have been downDom

**Comment by bradcray** [comment]
Oops, totally makes sense.  Sorry that I didn't think of that.

Thinking just enough to probably be dangerous:  What does it mean that ArrayViewSlice.chpl doesn't support this dsi routine?  It looks like the inherited version returns false, but my intuition is that it would need a similar method to this one?

**Comment by e-kayrakli** [comment]
> Thinking just enough to probably be dangerous: What does it mean that ArrayViewSlice.chpl doesn't support this dsi routine? It looks like the inherited version returns false, but my intuition is that it would need a similar method to this one?

This method is only used on domains, and we don't have an `ArrayViewSliceDom` type. That's why it's not applicable there.

In other words, we'd call this method on the domain we create under the hood for the slice:

```chapel
var A = newBlockArr({1..10});

ref ASlice = A[2..9];  // I hope I am not mistaken, but we create a BlockDom in this scenario

forall i in ASlice.domain {
   ... A[i] ...
}
```

**Comment by bradcray** [comment]
> This method is only used on domains, and we don't have an ArrayViewSliceDom type.

Ah yes, of course.  I was reading too quickly and just figured that the method was on the array type.

**Pull request merged**